### PR TITLE
Fix pager sort to ignore case and consistent ruby-vs-sql

### PR DESCRIPTION
**Pull request by pkarman**
If you ask Ruby which comes first alphabetically, `Newark` or `New York`, you'll get a different answer than if you ask PostgreSQL the same question. Substituting an underscore for a space in Ruby makes them sort the same way, predictably.

### CODE REVIEW

File: app/models/task_sorter.rb [hunk 1]
```diff
@@ -31,8 +31,24 @@ def sorted_tasks
...
+    return false if col =~ /^is_/
+    return false if col =~ /_count$/
+
+    true
+  end
+
   def order_clause
-    column.sorting_columns.map { |col| "#{column.sorting_table}.#{col} #{sort_order}" }.join(", ")
+    clauses = column.sorting_columns.map do |col|
+      tbl_clause = if sort_requires_case_norm?(col)
```
Comment (lowellrex) [inline app/models/task_sorter.rb]: What's the disadvantage in always normalizing casing?

Comment (pkarman) [reply app/models/task_sorter.rb]: some columns do not support it: timestamps, integers, booleans. if we put `UPPER` on everything we get psql errors.

File: app/models/task_sorter.rb [hunk 2]
```diff
@@ -31,8 +31,24 @@ def sorted_tasks
 
   private
 
+  def sort_requires_case_norm?(col)
+    return false if col =~ /_at$/
+    return false if col =~ /^is_/
+    return false if col =~ /_count$/
```
Comment (lowellrex) [inline app/models/task_sorter.rb]: Can we add a code comment here that explains that we're using column name as a heuristic for the data type?

File: app/models/task_sorter.rb (hunk 1 above)
Comment (tomas-nava) [reply app/models/task_sorter.rb]: Do these regexes capture all the timestamps, integers, and booleans that'll come through the task sorter? We definitely have columns of all three types that are not named according to those patterns.

Comment (pkarman) [reply app/models/task_sorter.rb]: they cover all the use cases we currently test. I kept adding them till all the tests in `spec/models/task_sorter_spec.rb` passed.

**Pull request merged**